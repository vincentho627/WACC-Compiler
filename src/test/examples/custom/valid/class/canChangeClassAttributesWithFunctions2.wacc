# Simple Class instantiation

begin
  class Vehicle is
    public string name ;
    public int tyres 
    public string getName() is 
      return name
    end
    public int getNumber() is
      return tyres
    end
    public int setNumber() is
      tyres = tyres * 5 ;
      return tyres
    end
  end
  Vehicle bus = new Vehicle("Bus", 6) ;
  Vehicle car = new Vehicle("Car", 4) ;
  println bus.tyres ;
  println car.tyres ;
  string v = bus.name ;
  println v 
end








