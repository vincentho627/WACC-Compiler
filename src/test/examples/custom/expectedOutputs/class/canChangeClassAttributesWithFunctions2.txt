6
4
Bus